iceoryx_dust/include/iceoryx_dust/internal/singleton.inl

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_DUST_SINGLETON_INL
+#define IOX_DUST_SINGLETON_INL
+
+#include "iceoryx_dust/singleton.hpp"
+
+namespace iox
+{
+
+template <typename T>
+bool Singleton<T>::isInitialized()
+{
+    return ptr().load(std::memory_order_relaxed) != nullptr;
+}
+
+template <typename T>
+template <typename... Args>
+T& Singleton<T>::init(Args&&... args)
+{
+    std::lock_guard<std::mutex> g(lock());
+    if (!isInitialized())
```

Comment:
Shouldn't we report this with an error? I would return here an `cxx::expected<>` with an error enum stating that it is already initialized. What when I initialize the singleton again with some other arguments then I may wrongly assume that this is initialized differently. Especially since there is a race condition:
```cxx
if ( !Singleton::isInitialized() ) {
   // if this is called by multiple threads with different parameters some arbitrary thread wins
  Singleton::init(1,2,3);  // in init only one gets the lock the others are then dismissed feels wrong.
}
```

I know that you defined the behavior in the doxygen comment. But I would bet that during a review nearly no developer reads the full documentation of all function calls which they are reviewing and then such a bug sneaks in faster than one thinks.